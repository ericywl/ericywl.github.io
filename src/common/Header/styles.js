import styled from "styled-components"

export const SideLinks = styled.div`
  a {
    color: #6d6d6d;
    font-size: 14pt;
    text-decoration: none;
    margin-right: 1rem;
    padding-bottom: 5px;

    &:last-child {
      margin-right: unset;
    }

    &:hover {
      border-bottom: 3px solid black;
      transition: 0.2s;
    }
  }

  @media (max-width: 960px) {
    display: none;
  }
`

export const Wrapper = styled.div`
  background: transparent;
  width: 100%;
`

export const Navbar = styled.div`
  padding: 1.5rem 0;
  display: flex;
  align-items: center;
  justify-content: space-between;

  a {
    color: #212121;
  }
`
