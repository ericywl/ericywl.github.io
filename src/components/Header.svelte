<script lang="ts">
    export let y: number;
    let linkedin = "https://linkedin.com/in/ericywl";

    let tabs = [
        {
            name: "Projects",
            link: "#projects",
        },
        {
            name: "About Me",
            link: "#about",
        },
    ];
</script>

<header
    class={"sticky z-[10] top-0 duration-200 px-6 flex items-center justify-between border border-solid " +
        (y > 0
            ? "py-4 bg-slate-950 border-violet-950"
            : "py-6 bg-transparent border-transparent")}
>
    <h1 class="font-medium">
        <b class="font-bold poppins">Eric</b> Yap
    </h1>
    <div class="sm:flex ml-auto pr-4 items-center gap-4 hidden">
        {#each tabs as tab}
            <a href={tab.link} class="duration-200 hover:text-violet-400">
                <p>{tab.name}</p>
            </a>
        {/each}
    </div>
    <a
        href={linkedin}
        target="_blank"
        class="blue-shadow relative overflow-hidden px-5 py-2 group rounded-full
        bg-white text-slate-950"
    >
        <div
            class="absolute top-0 right-full w-full h-full bg-violet-400 opacity-20 group-hover:translate-x-full z-0 duration-200"
        />
        <h4 class="relative z-9">Get In Touch</h4>
    </a>
</header>
