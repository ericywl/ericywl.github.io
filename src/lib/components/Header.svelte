<script lang="ts">
    import { base } from "$app/paths";
    import { myLinkedin } from "../data/links";
    import { type Tab } from "../data/models";

    export let scrollY: number;

    let tabs: Tab[] = [
        {
            name: "Projects",
            link: "#projects",
            external: false,
        },
        {
            name: "About Me",
            link: "#about",
            external: false,
        },
        {
            name: "Resume",
            link: base + "/ericywl_resume.pdf",
            external: true,
        },
    ];

    let showDropdown: boolean = false;
    function closeDropdown(e: Event) {
        showDropdown = false;
        e.stopPropagation();
    }

    function toggleShowDropdown(e: Event) {
        showDropdown = !showDropdown;
        e.stopPropagation();
    }
</script>

<header
    class={"sticky z-[10] top-0 duration-200 px-6 flex items-center justify-between border border-solid " +
        (scrollY > 0
            ? "py-4 bg-slate-950 border-violet-950"
            : "py-6 bg-transparent border-transparent")}
>
    <div class="flex">
        <div class="max-sm:block hidden">
            <button
                on:click={toggleShowDropdown}
                class="pr-4 focus:outline-none focus:border-white"
            >
                <i class="fa-solid fa-bars" />
            </button>
            <div
                class={"absolute z-10 border border-violet-700 mt-6 -ml-6 px-2 rounded-md py-2 w-full " +
                    "blue-shadow bg-opacity-65 bg-violet-950 duration-300 transition-all overflow-hidden " +
                    (showDropdown ? "max-h-40" : "max-h-0 invisible")}
            >
                {#each tabs as tab}
                    {#if tab.external}
                        <a
                            href={tab.link}
                            target="_blank"
                            class="block px-4 py-2"
                        >
                            <p>{tab.name}</p>
                        </a>
                    {:else}
                        <a href={tab.link} class="block px-4 py-2">
                            <p>{tab.name}</p>
                        </a>
                    {/if}
                {/each}
            </div>
        </div>
        <a href={base} class="hover:text-violet-400 hover:animate-pulse">
            <h1 class="font-medium">
                <b class="font-bold poppins">Eric</b> Yap
            </h1>
        </a>
    </div>
    <div class="sm:flex ml-auto pr-4 items-center gap-4 hidden">
        {#each tabs as tab}
            {#if tab.external}
                <a
                    href={tab.link}
                    target="_blank"
                    class="duration-200 hover:text-violet-400"
                >
                    <p>{tab.name}</p>
                </a>
            {:else}
                <a href={tab.link} class="duration-200 hover:text-violet-400">
                    <p>{tab.name}</p>
                </a>
            {/if}
        {/each}
    </div>
    <a
        href={myLinkedin}
        target="_blank"
        class="blue-shadow relative overflow-hidden px-5 py-2 group rounded-full
        bg-white text-slate-950"
    >
        <div
            class="absolute top-0 right-full w-full h-full bg-violet-400 opacity-20 group-hover:translate-x-full z-0 duration-200"
        />
        <h4 class="relative z-9">Get In Touch</h4>
    </a>
</header>

<svelte:window on:click={closeDropdown} on:scroll={closeDropdown} />
