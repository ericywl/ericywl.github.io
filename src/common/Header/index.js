import React from "react"
import PropTypes from "prop-types"
import { Link } from "gatsby"
import AnchorLink from "react-anchor-link-smooth-scroll"

import { Wrapper, Navbar, SideLinks } from "./styles"
import { Container } from "Common/Container"
import icon from "Static/icons/robot.png"

const Header = ({ notFound }) => (
  <Wrapper>
    <Navbar as={Container}>
      <Link to="/">
        <img src={icon} width="70px" alt="Icon" />
      </Link>
      {!notFound ? (
        <SideLinks>
          <AnchorLink href="#about">About</AnchorLink>
          <AnchorLink href="#projects">Projects</AnchorLink>
          <a
            href="/ericywl_resume.pdf"
            target="_blank"
            rel="noopener noreferrer"
          >
            Resume
          </a>
        </SideLinks>
      ) : (
        ""
      )}
    </Navbar>
  </Wrapper>
)

Header.propTypes = {
  notFound: PropTypes.bool,
}

export default Header
