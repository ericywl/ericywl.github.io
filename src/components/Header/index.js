import React from "react"
import PropTypes from "prop-types"
import { Link } from "gatsby"
import AnchorLink from "react-anchor-link-smooth-scroll"

import { Wrapper, Navbar, SideLinks } from "./styles"
import { title } from "Data"
import { Container } from "Common/Container"
import icon from "Static/icons/robot.png"

const Header = () => (
  <Wrapper>
    <Navbar as={Container}>
      <Link to="/"><img src={icon} width="70px" alt="Icon" /></Link>
      <SideLinks>
        <AnchorLink href="#about">About</AnchorLink>
        <AnchorLink href="#projects">Projects</AnchorLink>
      </SideLinks>
    </Navbar>
  </Wrapper>
)

export default Header
