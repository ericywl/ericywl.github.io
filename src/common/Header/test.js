import React from "react"
import renderer from "react-test-renderer"

import Header from "./index"
import { SideLinks } from "./styles"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Header notFound={false} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  it("renders SideLink when in normal pages", () => {
    const component = renderer.create(<Header notFound={false} />).root
    expect(component.findAllByType(SideLinks).length).toBe(1)
  })
  
  it("does not render SideLink when in NotFound page", () => {
    const component = renderer.create(<Header notFound={true} />).root
    expect(component.findAllByType(SideLinks).length).toBe(0)
  })
})
