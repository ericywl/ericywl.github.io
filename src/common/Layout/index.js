import React from "react"
import PropTypes from "prop-types"

import { Wrapper, Global } from "./styles"
import Header from "Common/Header"
import Footer from "Common/Footer"
import "./fonts.css"

const Layout = ({ children, noAnchor }) => (
  <Wrapper>
    <Header noAnchor={noAnchor} />
    <Global />
    {children}
    <Footer />
  </Wrapper>
)

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  noAnchor: PropTypes.bool,
}

export default Layout
