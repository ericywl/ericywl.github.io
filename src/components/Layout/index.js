import React from "react"
import PropTypes from "prop-types"
import { useStaticQuery, graphql } from "gatsby"

import { Container, Global } from "./styles"
import Header from "../Header"
import "./fonts.css"

const Layout = ({ children }) => {
  return (
    <Container>
      <Header />
      <Global />
      {children}
      <footer>Â© All rights reserved, {new Date().getFullYear()}</footer>
    </Container>
  )
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
}

export default Layout
