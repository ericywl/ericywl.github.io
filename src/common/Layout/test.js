import React from "react"
import renderer from "react-test-renderer"

import Layout from "./index"
import Header from "Common/Header"

const Dummy = () => <div>Dummy child</div>

describe("Layout", () => {
  it("renders correctly", () => {
    const component = renderer.create(
      <Layout children={<Dummy />} noAnchor={false} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("passes noAnchor to Header", () => {
    const falseComponent = renderer.create(
      <Layout children={<Dummy />} noAnchor={false} />
    )
    expect(falseComponent.root.findByType(Header).props.noAnchor).toBe(false)

    const trueComponent = renderer.create(
      <Layout children={<Dummy />} noAnchor={true} />
    )
    expect(trueComponent.root.findByType(Header).props.noAnchor).toBe(true)
  })
})
