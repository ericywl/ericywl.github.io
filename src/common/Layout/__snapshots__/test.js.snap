// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  <div
    className="sc-bwzfXH xQRDH"
  >
    <div
      className="sc-bxivhb KOyPd sc-htpNat gcOxHz"
    >
      <a
        href="/"
      >
        <img
          alt="Icon"
          src="test-file-stub"
          width="70px"
        />
      </a>
      <div
        className="sc-bdVaJa dGfGoS"
      >
        <a
          href="#about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          href="#projects"
          onClick={[Function]}
        >
          Projects
        </a>
        <a
          href="/ericywl_resume.pdf"
          rel="noopener noreferrer"
          target="_blank"
        >
          Resume
        </a>
      </div>
    </div>
  </div>,
  <div>
    Dummy child
  </div>,
  <div
    className="sc-ifAKCX beCMfH"
  >
    <div
      className="sc-bxivhb KOyPd sc-EHOje cnSYZS"
    >
      <div
        className="sc-gzVnrw bVysae"
      >
        <h2>
          Eric Yap
        </h2>
        <span>
          Â© All rights reserved, 
          2019
           | Made with GatsbyJS
        </span>
      </div>
      <div
        className="sc-bZQynM bGniFE"
      >
        <a
          href="https://linkedin.com/in/ericywl"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="LinkedIn"
            src="/icons/linkedin.svg"
            width="24"
          />
        </a>
        <a
          href="https://github.com/ericywl"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="GitHub"
            src="/icons/github.svg"
            width="24"
          />
        </a>
      </div>
    </div>
  </div>,
]
`;
